wasm-crypto/src/lib.rs

```diff
@@ -475,6 +476,38 @@ pub fn encode_input_for_account_outpoint(
     Ok(input.encode())
 }
 
+#[wasm_bindgen]
+pub fn estimate_transaction_size(
+    inputs: &[u8],
+    mut opt_utxos: &[u8],
+    mut outputs: &[u8],
+) -> Result<usize, Error> {
+    let mut tx_outputs = vec![];
+    while !outputs.is_empty() {
+        let output = TxOutput::decode(&mut outputs).map_err(|_| Error::InvalidOutput)?;
+        tx_outputs.push(output);
+    }
+
+    let size = tx_size_with_outputs(&tx_outputs);
+    let inputs_size = inputs.len();
+
+    let mut total_size = size + inputs_size;
+
+    while !opt_utxos.is_empty() {
+        let utxo = Option::<TxOutput>::decode(&mut opt_utxos).map_err(|_| Error::InvalidInput)?;
```

Comment:
This also misses the encoded length of the witness vector.Not sure what sort of precision is needed here. However for example in case of fee estamation, it may not be good to underestimate size if blocks are close to empty and the user wants to pay the minimum relay feerate.

Reply:
had me worried for a second why is the test passing if the vector length is not included,
but it is included in the `tx_size_with_outputs` function, it creates an empty signed transaction so all of the vectors are included there, inputs and witnesses are just later added.
The nodejs test checks that the estimated size is the same as the real encoded signed transaction